Addressing issue 89: incorporating SUP on App Registry as OCI Registry (#123)
margo/specification: system-design/concepts/workloads/application-registry.md

```mermaid
flowchart
   A[WFM, or internal Application Catalog] -- Application Descriptions link to --> B[Component Registry] 
   C[Application Registry] -- Application Descriptions link to --> B
   B -- hosted Components links to --> D[Container Registry]
   A -->|pulls Application Package | C
   F[App Developer] -->|uploads Application Package to| C
   G["Marketplace"] -- points to Application Package --> C
   C -->|hosts 0..*| E@{ shape: docs, label: "Application Packages"} 
   C -->|validates token| H[Authentication Service] 
   A -->|requests token| H
   style H stroke-dasharray: 3 6

   style B fill:#ABC
   style C fill:#ABC
   style D fill:#ABC
   style G fill:#ABC
```

As shown in the figure above, an `Application Developer` uploads an [Application package](application-package.md) to an Application Registry. From there, it is available to a `Workload Fleet Manager` (WFM).
The WFM acts as a client to pull an [Application Package](application-package.md) from the Application Registry. It would then list this Application Package on its UI (e.g., an internal `Application Catalog`) to enable the usage on its managed devices.

> An `Authentication Service` manages access control for the Application Registry. Hence, the WFM requests a token from there to include it into its requests to the Application Registry. The received token is then validated by the Application Registry through interaction with the Authentication Service.

Reviewer: ```suggestion
An `Authentication Service` manages access control for the Application Registry. The WFM requests a token from the Authentication Service to include in the requests to the Application Registry. The received token is then validated by the Application Registry through interaction with the Authentication Service.
```

Author: done

> The Application Registry's API is compliant with the [OCI Registry API (v1.1.0)](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md). A hosted Application Package is provided by listing its parts as layers in an [image manifests](https://github.com/opencontainers/image-spec/blob/v1.0.1/manifest.md)) that can be requested through the API.

Reviewer: ```suggestion
The Application Registry's API is compliant with the [OCI Registry API (v1.1.0)](https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md). A hosted Application Package is provided by listing its parts as layers in an [image manifests](https://github.com/opencontainers/image-spec/blob/v1.0.1/manifest.md) that can be requested through the API.
```

Author: done
